--- /home/dministrator/my-img/patches/sd_cpp_backup/include/stable-diffusion.h	2026-04-30 15:51:55.762743322 +0800
+++ include/stable-diffusion.h	2026-04-30 16:57:03.812338437 +0800
@@ -316,6 +316,26 @@
 } sd_hires_params_t;
 
 typedef struct {
+    bool enabled;
+    float b1;
+    float b2;
+    float s1;
+    float s2;
+} sd_freeu_params_t;
+
+typedef struct {
+    bool enabled;
+    float scale;
+} sd_sag_params_t;
+
+typedef struct {
+    bool enabled;
+    float percentile;
+    float mimic_scale;
+    float threshold_percentile;
+} sd_dynamic_cfg_params_t;
+
+typedef struct {
     const sd_lora_t* loras;
     uint32_t lora_count;
     const char* prompt;
@@ -339,6 +359,9 @@
     sd_tiling_params_t vae_tiling_params;
     sd_cache_params_t cache;
     sd_hires_params_t hires;
+    sd_freeu_params_t freeu;
+    sd_sag_params_t sag;
+    sd_dynamic_cfg_params_t dynamic_cfg;
 } sd_img_gen_params_t;
 
 typedef struct {
