--- /home/dministrator/my-img/patches/sd_cpp_backup/include/stable-diffusion.h	2026-04-30 15:51:55.762743322 +0800
+++ /opt/stable-diffusion.cpp/include/stable-diffusion.h	2026-04-30 15:55:34.933336248 +0800
@@ -314,6 +314,14 @@
     float denoising_strength;
     int upscale_tile_size;
 } sd_hires_params_t;
+typedef struct {
+    bool enabled;
+    float b1;
+    float b2;
+    float s1;
+    float s2;
+} sd_freeu_params_t;
+
 
 typedef struct {
     const sd_lora_t* loras;
@@ -339,6 +347,7 @@
     sd_tiling_params_t vae_tiling_params;
     sd_cache_params_t cache;
     sd_hires_params_t hires;
+    sd_freeu_params_t freeu;
 } sd_img_gen_params_t;
 
 typedef struct {
