--- /home/dministrator/my-img/patches/sd_cpp_backup/src/diffusion_model.hpp	2026-04-30 15:51:55.766409989 +0800
+++ src/diffusion_model.hpp	2026-04-30 16:10:09.373530816 +0800
@@ -29,6 +29,11 @@
     const sd::Tensor<float>* vace_context             = nullptr;
     float vace_strength                               = 1.f;
     const std::vector<int>* skip_layers               = nullptr;
+    bool freeu_enabled = false;
+    float freeu_b1 = 1.3f;
+    float freeu_b2 = 1.4f;
+    float freeu_s1 = 0.9f;
+    float freeu_s2 = 0.2f;
 };
 
 template <typename T>
@@ -107,6 +112,7 @@
         GGML_ASSERT(diffusion_params.x != nullptr);
         GGML_ASSERT(diffusion_params.timesteps != nullptr);
         static const std::vector<sd::Tensor<float>> empty_controls;
+        unet.set_freeu(diffusion_params.freeu_enabled, diffusion_params.freeu_b1, diffusion_params.freeu_b2, diffusion_params.freeu_s1, diffusion_params.freeu_s2);
         return unet.compute(n_threads,
                             *diffusion_params.x,
                             *diffusion_params.timesteps,
