--- /home/dministrator/my-img/patches/sd_cpp_backup/src/stable-diffusion.cpp	2026-04-30 15:51:55.770076655 +0800
+++ /opt/stable-diffusion.cpp/src/stable-diffusion.cpp	2026-04-30 16:07:36.743760441 +0800
@@ -147,6 +147,11 @@
     bool use_pmid                        = false;
 
     bool is_using_v_parameterization     = false;
+    bool freeu_enabled = false;
+    float freeu_b1 = 1.3f;
+    float freeu_b2 = 1.4f;
+    float freeu_s1 = 0.9f;
+    float freeu_s2 = 0.2f;
     bool is_using_edm_v_parameterization = false;
 
     std::map<std::string, ggml_tensor*> tensors;
@@ -1642,6 +1647,11 @@
             diffusion_params.guidance           = &guidance_tensor;
             diffusion_params.ref_latents        = &ref_latents;
             diffusion_params.increase_ref_index = increase_ref_index;
+            diffusion_params.freeu_enabled = this->freeu_enabled;
+            diffusion_params.freeu_b1 = this->freeu_b1;
+            diffusion_params.freeu_b2 = this->freeu_b2;
+            diffusion_params.freeu_s1 = this->freeu_s1;
+            diffusion_params.freeu_s2 = this->freeu_s2;
             diffusion_params.controls           = &controls;
             diffusion_params.control_strength   = control_strength;
             diffusion_params.vace_context       = vace_context.empty() ? nullptr : &vace_context;
@@ -3345,6 +3355,13 @@
     sd_ctx->sd->sampler_rng->manual_seed(request.seed);
     sd_ctx->sd->set_flow_shift(sd_img_gen_params->sample_params.flow_shift);
     sd_ctx->sd->apply_loras(sd_img_gen_params->loras, sd_img_gen_params->lora_count);
+    sd_ctx->sd->freeu_enabled = sd_img_gen_params->freeu.enabled;
+    if (sd_img_gen_params->freeu.enabled) {
+        sd_ctx->sd->freeu_b1 = sd_img_gen_params->freeu.b1;
+        sd_ctx->sd->freeu_b2 = sd_img_gen_params->freeu.b2;
+        sd_ctx->sd->freeu_s1 = sd_img_gen_params->freeu.s1;
+        sd_ctx->sd->freeu_s2 = sd_img_gen_params->freeu.s2;
+    }
 
     ImageVaeAxesGuard axes_guard(sd_ctx, sd_img_gen_params, request);
 
