diff --git a/examples/common/common.h b/examples/common/common.h
index 5afe89b..b764300 100644
--- a/examples/common/common.h
+++ b/examples/common/common.h
@@ -190,6 +190,13 @@ struct SDGenerationParams {
     int upscale_repeats   = 1;
     int upscale_tile_size = 128;
 
+    // HiRes Fix parameters
+    bool hires_fix        = false;
+    int hires_width       = 0;
+    int hires_height      = 0;
+    float hires_strength  = 0.5f;
+    int hires_steps       = 20;
+
     std::map<std::string, float> lora_map;
     std::map<std::string, float> high_noise_lora_map;
 
